src/bijectors/ordered.jl

```diff
@@ -0,0 +1,60 @@
+"""
+    OrderedBijector()
+
+A bijector mapping ordered vectors in ℝᵈ to an unordered vectors in ℝᵈ.
+
+## See also
+- [Stan's documentation](https://mc-stan.org/docs/2_27/reference-manual/ordered-vector.html)
+  - Note that this transformation and its inverse are the _opposite_ of in this reference.
+"""
+struct OrderedBijector <: Bijector{1} end
+
+"""
+    ordered(d::Distribution)
+
+Returns a `Distribution` whose domain is now ordered vectors.
+"""
+ordered(d::Distribution) = Bijectors.transformed(d, OrderedBijector())
+
+(b::OrderedBijector)(y::AbstractVecOrMat) = _transform_ordered(y)
+
+function _transform_ordered(y::AbstractVector)
+    x = similar(y)
+    x[1] = y[1]
+    @inbounds for i = 2:size(x, 1)
+        x[i] = x[i - 1] + exp(y[i])
+    end
```

Comment:
An alternative would be something like
```julia
x = map(exp, y)
x[begin] = y[begin]
cumsum!(x, x)
```
(or a `map` that directly avoids `exp` of the first element). Could be interesting to benchmark as well, in particular since `cumsum!` accepts a `dims` keyword argument.

Reply:
`begin` isn't supported on 1.3 though, right?

Reply:
I'll give that a go :+1:

I also started doing `map(exp, y)` and `x[1] = y[1]` after, but this doesn't work in the _inverse_ since we're using `log`. BUT that's not a good reason not to do it in this case; I just tried doing it everywhere, failed and I just reverted the change without giving it too much thought (just noted why it failed) :upside_down_face:

Reply:
Quick benchmark: for the batched-version, it's slower or the same as the for-loop approach :confused: